src/main/kotlin/org/rust/cargo/runconfig/buildtool/CargoBuildManager.kt

```diff
@@ -278,15 +277,23 @@ object CargoBuildManager {
     }
 
     private val cargoBuildPatch: CargoPatch = { commandLine ->
-        val additionalArguments = commandLine.additionalArguments.toMutableList()
-        additionalArguments.remove("-q")
-        additionalArguments.remove("--quiet")
-        addFormatJsonOption(additionalArguments, "--message-format")
+        val additionalArguments = mutableListOf<String>().apply {
+            addAll(commandLine.additionalArguments)
+            remove("-q")
+            remove("--quiet")
+            val format = if (SystemInfo.isUnix) "json-diagnostic-rendered-ansi" else "json"
+            addFormatJsonOption(this, "--message-format", format)
+        }
+
+        val oldVariables = commandLine.environmentVariables
         val environmentVariables = EnvironmentVariablesData.create(
-            EnvironmentUtil.getEnvironmentMap() +
-                commandLine.environmentVariables.envs - "CI" + ("TERM" to "ansi"),
-            false
+            oldVariables.envs + mapOf(
+                "CARGO_TERM_PROGRESS_WHEN" to "always",
+                "CARGO_TERM_PROGRESS_WIDTH" to "1000"
```

Comment:
Is there documentation about these envs?

Reply:
Yes, https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables

Reply:
Could you add this link here, please? Just to understand what they mean
Also, it would be great to add a short comment why we need them at all